tests/src/utils/test_rlp.py

```diff
@@ -1,25 +1,17 @@
-import random
-
 import pytest
+from hypothesis import given
+from hypothesis.strategies import binary, lists, recursive
 from rlp import codec, decode, encode
 
 from tests.utils.constants import TRANSACTIONS
-from tests.utils.helpers import flatten, rlp_encode_signed_data
+from tests.utils.errors import cairo_error
+from tests.utils.helpers import rlp_encode_signed_data
 
 
 class TestRLP:
     class TestDecodeType:
-        @pytest.mark.parametrize(
-            "payload_len",
-            [0, 55, 256],
-        )
-        def test_should_match_decoded_rlp_type_string(self, cairo_run, payload_len):
-            # generate random string of bytes. if payload_len is 0, then generate a single byte inferior to 0x80
-            data = (
-                random.randbytes(payload_len)
-                if payload_len != 0
-                else random.randint(0, 0x80)
-            )
+        @given(data=lists(binary()) | binary())
```

Comment:
is this generating a random sequence of bytes ?

Reply:
either random bytes or a list of random bytes, ie either `b'1234'` or `[b'1234']`